### DISCUSSION TITLE
support of BLE sensor CGDK2 (T/H) possible?

### DISCUSSION THREAD

**Discussion by michapr** [arendst/Tasmota#17370] [Ideas]
Hi,
I want to switch from OpenMQTTGateway to Tasmota and just miss the support for this sensor.

I get in log 
`Dec 12 17:45:45 tasmota-190F70-3952 ESP-M32 582D3411915D: Payload decrypt failed`

Have got the Key, added, but still the same message
`{"MIKeys":{"582D3411915D":"92E96A650F4xxxxxxxx339C8718FAABB"}}`
(replaced the "xx" here only)

Do I something wrong, or does the message not say that device is supported, but value is still decrypted?

If not supported, what can I do for adding it? Write down the encoding of the message here?

EDIT: looks like not supported just now, right (second: "unknown")?
![grafik](https://user-images.githubusercontent.com/11854556/207111184-f94d0508-c1e7-46bf-aa06-1805d461f097.png)

Thanks!

**Comment by barbudor** [comment]
There seems to be 2 independant issues here:
- failed decryption
- CGDK2 not supported (yet)

Even if CGDK2  is not supported you shouldn't get that message if the key was correct
Until you get a correct key it will not be possible to work on supporting it

**Reply by sfromis** [reply]
More generally, you should not expect BLE in Tasmota to work for other devices than listed in the docs.

**Reply by michapr** [reply]
>     * failed decryption
> 
>     * CGDK2 not supported (yet)
> 
> 
> Even if CGDK2 is not supported you shouldn't get that message if the key was correct Until you get a correct key it will not be possible to work on supporting it

You are right, I have used the wrong key (have several sensors of same type and have tried to use two if them using the registration and key.

**BUT:**
For this sensor no key is required! 
I'm using it in OpenMQTTGateway (with own code snippet) since 2 years about.
The "trick" is to activate the sensor in Qingping App (NOT MiHome App !) - then the sensor will report all values without key.

Here is the encoding with sample data:

```
Service Data: UUID 0xFDCD
Data:
first device:
88 10 5d 91 11 34 2d 58 01 04 9f 00 d0 01 02 01 42
88 10 5d 91 11 34 2d 58 01 04 9f 00 c7 01 02 01 42

second device:
88 10 08 92 11 34 2d 58 01 04 a3 00 d4 01 02 01 54
88 10 08 92 11 34 2d 58 01 04 a3 00 c0 01 02 01 54
```

![grafik](https://user-images.githubusercontent.com/11854556/207385588-d4845111-04cf-469e-b479-f996a77daa49.png)

Maybe somebody could integrate it?
Until then, I will have to continue using OpenMQTTGateway ;)

Thanks!

**Comment by michapr** [comment]
Can this be integrated?
Or what could I do to help?

Thanks!

**Reply by Jason2866** [reply]
Could be integrated, if someone is doing it. Tasmota is Open Source, none is paid for anything.
So if you like to have "something in", you have to find someone who finds this interesting and is implementing.

**Reply by michapr** [reply]
>  So if you like to have "something in", you have to find someone who finds this interesting and is implementing.

That's why asking here ;)
